### PR TITLE
Update Docker Documentation

### PR DESCRIPTION
**Pull request by Gamenot**
Fix some missing steps and clarify some existing steps for getting an example running.

This was brought up by someone attempting to use a docker image.

#472

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -228,11 +228,19 @@ If you're comfortable using docker or are on a platform without suitable support
 
 ```bash
 docker run --rm -it -v $(PWD):/src -p 8081:8081 huaweinoah/smarts:<version>
-# E.g. docker run --rm -it -v $(PWD):/src -p 8081:8081 huaweinoah/smarts:v0.4.3-pre
+# E.g. docker run --rm -it -v $(PWD):/src -p 8081:8081 huaweinoah/smarts:v0.4.12
```
Comment (Adaickalavan) [inline README.md:231]:
```suggestion
# E.g. docker run --rm -it -v $PWD:/src -p 8081:8081 huaweinoah/smarts:v0.4.12
```

File: README.md [hunk 2]
```diff
@@ -228,11 +228,19 @@ If you're comfortable using docker or are on a platform without suitable support
 
 ```bash
 docker run --rm -it -v $(PWD):/src -p 8081:8081 huaweinoah/smarts:<version>
```
Comment (Adaickalavan) [inline README.md:230]:
```suggestion
docker run --rm -it -v $PWD:/src -p 8081:8081 huaweinoah/smarts:<version>
```

File: README.md [hunk 3]
```diff
@@ -228,11 +228,19 @@ If you're comfortable using docker or are on a platform without suitable support
...
+
+# Build an example
+# This needs to be done the first time and after changes to the example
+$ scl scenario build scenarios/loop --clean
+
+# Run an example
+# add --headless if you do not need visualisation
 $ python examples/single_agent.py scenarios/loop
 
 # On your most machine visit http://localhost:8081 to see the running simulation in
```
Comment (Adaickalavan) [inline README.md:246]:
```suggestion
# On your host machine visit http://localhost:8081 to see the running simulation in
```

**Review by Adaickalavan — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -227,15 +227,24 @@ python examples/run_smarts.py --algo SAC --scenario ./scenarios/loop --n_agents
...
+# E.g. docker run --rm -it -v $PWD:/src -p 8081:8081 huaweinoah/smarts:v0.4.12
 # <press enter>
 
 # Run Envision server in the background
+# This will only need to be run if you want visualisation
 $ scl envision start -s ./scenarios -p 8081 &
+
+# Build an example
+# This needs to be done the first time and after changes to the example
+$ scl scenario build scenarios/loop --clean
```
Comment (JingfeiPeng) [inline README.md:241]: Are the users supposed to enter the docker container to execute these commands? if so we can provide the lines for entering docker container as well

Comment (Gamenot) [reply README.md:241]: `$ docker run --rm -it -v $PWD:/src -p 8081:8081 huaweinoah/smarts:<version>`

Starts the interactive console, we assume that they have ran that.

**Review by JingfeiPeng — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -227,15 +227,24 @@ python examples/run_smarts.py --algo SAC --scenario ./scenarios/loop --n_agents
...
-docker run --rm -it -v $(PWD):/src -p 8081:8081 huaweinoah/smarts:<version>
-# E.g. docker run --rm -it -v $(PWD):/src -p 8081:8081 huaweinoah/smarts:v0.4.3-pre
+$ cd /path/to/SMARTS
+$ docker run --rm -it -v $PWD:/src -p 8081:8081 huaweinoah/smarts:<version>
+# E.g. docker run --rm -it -v $PWD:/src -p 8081:8081 huaweinoah/smarts:v0.4.12
 # <press enter>
 
 # Run Envision server in the background
+# This will only need to be run if you want visualisation
 $ scl envision start -s ./scenarios -p 8081 &
```
Comment (JingfeiPeng) [inline README.md:237]: do we need the `&` at the end?

Comment (Gamenot) [reply README.md:237]: In this case, yes.

**Pull request merged**